planner/core/logical_plan_builder.go

```diff
@@ -272,6 +272,16 @@ func (p *LogicalJoin) setPreferredJoinType(hintInfo *tableHintInfo) error {
 	return nil
 }
 
+func resetNotNullFlag(schema *expression.Schema, start, end int) {
+	for i := start; i < end; i++ {
+		col := *schema.Columns[i]
+		newFieldType := *col.RetType
+		newFieldType.Flag &= ^mysql.NotNullFlag
+		col.RetType = &newFieldType
+		schema.Columns[i] = &col
```

Comment:
Why do we update the pointer? would it change?

Reply:
We first get the value of column 
```
col := *schema.Columns[i] 
```
and then update the flag of `col`, finally we assign the address of the newly created column to `schema.Columns[i] = &col`.